src/main/routes/weapon.resource.js

```diff
@@ -12,66 +13,29 @@ export default function() {
         "xScope": 10,
         "yScope": 5
     }*/
-    router.post('/', (req, res) => {
-        const scenarioID = req.body.scenario;
-        const existingActor = req.body.actor;
-        const newGun = req.body.name;
-        const newPower = req.body.power;
-        const newXscope = req.body.xScope;
-        const newYscope = req.body.yScope;
-        const actorExists = true;
-        const result = {
-            scenario: scenarioID,
-            actors: [{
-                'name': existingActor,
-                'type': 'PF Squad Soldier',
-                'health': 1,
-                'weapon': {
-                    'name': newGun,
-                    'power': newPower,
-                    'xScope': newXscope,
-                    'yScope': newYscope,
-                },
-                'position': {
-                    'xPos': 0,
-                    'yPos': 0
-                }
-            }, {
-                'name': 'RAS1',
-                'type': 'Rebel Army soldier',
-                'health': 1,
-                'weapon': {
-                    'name': 'rifle',
-                    'power': 1,
-                    'xScope': 1,
-                    'yScope': 1
-                },
-                'position': {
-                    'xPos': 100,
-                    'yPos': 0
-                }
-            }, {
-                'name': 'RAS2',
-                'type': 'Rebel Army soldier',
-                'health': 1,
-                'weapon': {
-                    'name': 'rifle',
-                    'power': 1,
-                    'xScope': 1,
-                    'yScope': 1
-                },
-                'position': {
-                    'xPos': 5,
-                    'yPos': 5
-                }
-            }]
-        };
+    router.post('/', async (req, res) => {
+        if (!req.body.scenario) res.send({status: 400, error: 'No scenario specidied'});
```

Comment:
There's a Typo here, use "No scenario was specified", instead of "No scenario **specidied**"